src/lsp_data.rs

```diff
@@ -85,11 +86,18 @@ pub mod ls_util {
         }
     }
 
-    pub fn range_from_vfs_file(_vfs: &Vfs, _fname: &Path) -> Range {
-        // FIXME: todo, endpos must be the end of the document, this is not correct
-
-        let end_pos = Position::new(0, 0);
-        Range{ start: Position::new(0, 0), end: end_pos }
+    pub fn range_from_vfs_file(_vfs: &Vfs, _fname: &Path) -> Result<Range, vfs::Error> {
+        let content = _vfs.load_file(_fname)?;
+        if content.is_empty() {
+            Ok(Range {start: Position::new(0, 0), end: Position::new(0, 0)})
+        } else {
+            // range is zero-based and the end position is exclusive
+            Ok(Range {
+                start: Position::new(0, 0),
+                end: Position::new(content.lines().count() as u64 - 1,
+                        content.lines().last().expect("String is not empty.").chars().count() as u64)
```

Comment:
The old implementation is (0, 0) for the given snippet, my suggestion should give (2, 0). Is it VS Code that errors or our code? If the later it might be nice to fix that code.